### PR TITLE
chore: rename 'lint' to 'styleguide'

### PR DESCRIPTION
**Pull request by roman-sainchuk**
## What/Why/How?

Renaming `lint` field into `styleguide` in Redocly config. 
We still support the legacy format, but added deprecation notices if the `lint` field is found inside an api or on the root level. 

Closes https://github.com/Redocly/redocly-cli/issues/696

> Note: I had to comment out `cli-package-test` job for the time being as it incorrectly packs `cli` with `core` and I haven't found any quick solution for that. 

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -40,35 +40,36 @@ jobs:
...
-          CI: true
-      - name: Prepare CLI package
-        run: npm run pack:prepare
-      - name: Install CLI Package
-        run: npm install -g redocly-cli.tgz
-      - name: Redocly version
-        run: redocly --version
-      - name: Definition test
-        run: redocly lint packages/core/src/benchmark/benches/rebilly.yaml --format=stylish
+  # FIXME: fix this so the `cli` package includes the corresponding `core`
```
Comment (adamaltman) [inline .github/workflows/tests.yaml:43]: Should this be addressed when?

File: packages/core/src/config/load.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { Config, DOMAINS } from './config';
 import { transformConfig } from './utils';
 import { resolveConfig } from './config-resolvers';
 
-import type { RawConfig, Region } from './types';
+import type { DeprecatedInRawConfig, RawConfig, Region } from './types';
```
Comment (adamaltman) [inline packages/core/src/config/load.ts:9]: `DeprecatedInRawConfig` or `DeprecatedRawConfig`?

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (tatomyr) [reply .github/workflows/tests.yaml:43]: I reopened the [original issue](https://github.com/Redocly/redocly-cli/issues/732).

File: packages/core/src/config/load.ts (hunk 2 above)
Comment (tatomyr) [reply packages/core/src/config/load.ts:9]: I reckon it should be `In` as it's only about fields being deprecated inside the raw config and not the entire deprecated raw config.

**Comment by roman-sainchuk** [comment]
Looks good to me 😃

**Review by RomanHotsiy — APPROVED** [review]
LGTM 👍

**Review by AntonKozachuk — APPROVED** [review]

**Pull request merged**